game-core/src/main/java/games/strategy/triplea/ui/unit/scroller/UnitScroller.java

```diff
@@ -163,8 +163,10 @@ private void drawUnitAvatarPane(final Territory t) {
     SwingUtilities.invokeLater(
         () -> {
           selectUnitImagePanel.removeAll();
-          selectUnitImagePanel.add(
-              avatarPanelFactory.buildPanel(matchedUnits, currentPlayerSupplier.get()));
+          if (currentPlayerSupplier.get() != null) {
+            selectUnitImagePanel.add(
+                avatarPanelFactory.buildPanel(matchedUnits, currentPlayerSupplier.get()));
```

Comment:
The logic might be more unified if we had a single list with an aggregated data type, and rendering was a no-op only if the given list was empty, rather than a no-op if either player is null, or list is empty. This seems fine though.